src/app/core/services/user/user.service.ts

```diff
@@ -78,29 +75,42 @@ export class UserService {
    */
   signInUserByToken(token?: string): Observable<CustomerLoginType> {
     if (token) {
-      return this.fetchCustomer({
-        headers: new HttpHeaders().set(ApiService.TOKEN_HEADER_KEY, token),
-      });
+      return this.fetchToken('refresh_token', { refresh_token: token }).pipe(switchMap(() => this.fetchCustomer()));
     } else {
       return this.fetchCustomer({ skipApiErrorHandling: true });
     }
   }
 
-  private fetchCustomer(options?: AvailableOptions): Observable<CustomerLoginType> {
+  fetchCustomer(options: AvailableOptions = {}): Observable<CustomerUserType> {
     return this.apiService.get<CustomerData>('customers/-', options).pipe(
       withLatestFrom(this.appFacade.isAppTypeREST$),
       concatMap(([data, isAppTypeRest]) =>
         forkJoin([
           isAppTypeRest && data.customerType === 'PRIVATE'
-            ? this.apiService.get<CustomerData>('privatecustomers/-')
+            ? this.apiService.get<CustomerData>('privatecustomers/-', options)
             : of(data),
-          this.apiService.get<{ pgid: string }>('personalization').pipe(map(data => data.pgid)),
+          this.apiService.get<{ pgid: string }>('personalization', options).pipe(map(data => data.pgid)),
         ])
       ),
       map(([data, pgid]) => ({ ...CustomerMapper.mapLoginData(data), pgid }))
     );
   }
 
+  fetchToken<T extends 'anonymous'>(grantType: T): Observable<TokenResponse>;
+  fetchToken<T extends GrantType, R extends FetchTokenOptions<T>>(grantType: T, options: R): Observable<TokenResponse>;
+  fetchToken<T extends GrantType, R extends FetchTokenOptions<T>>(
+    grantType: T,
+    options?: R
+  ): Observable<TokenResponse> {
+    return from(
+      this.oauthService.fetchTokenUsingGrant(
+        grantType,
+        options ?? {},
+        new HttpHeaders({ 'content-type': 'application/x-www-form-urlencoded' })
+      )
+    ).pipe(tap(tokenResponse => this.apiTokenService.setApiToken(tokenResponse.access_token)));
```

Comment:
I think this setting logic should be handled in the effect, don't you?
Because usually a `fetch` method doesn't involve any writing and it can become confusing.

For example, the `fetchAnonymousUser` effect simply calls this method and does nothing with the return value -> it would be more transparent if it wrote the return value to the `apiTokenService` itself, imo.

Also, you could save a dependency on the `ApiTokenService` here, making everything less entangled.